IBM/ibm-common-service-operator
# Get rid of security warning in preload.sh script

Issue: https://github.ibm.com/IBMPrivateCloud/roadmap/issues/58107

It is intended to avoid security warning when we apply MongoDB StatefulSet and its backup and restore Jobs

### How to test
1. Install CS MongoDB in one namespace as the backup location
2. Create a new namespace as the restore location for MongoDB
3. Execute `preload_data.sh` to copy the MongoDB data, and it should run successfully
```
✗ ./preload_data.sh --original-cs-ns keycloak --services-ns preload

[INFO] Copying mongodb from namespace keycloak to namespace preload
# Cleaning up any previous copy operations...
-----------------------------------------------------------------------
[✔] Previous run cleaned up.
# Deploying a temporary mongodb in preload
-----------------------------------------------------------------------
configmap/icp-mongodb-install created
configmap/icp-mongodb-init created
issuer.cert-manager.io/god-issuer created
configmap/ibm-cpp-config created
secret/icp-mongodb-admin created
certificate.cert-manager.io/icp-mongodb-client-cert created
configmap/icp-mongodb created
secret/icp-mongodb-keyfile created
secret/icp-mongodb-metrics created
serviceaccount/ibm-mongodb-operand created
service/mongodb created
service/icp-mongodb created
certificate.cert-manager.io/mongodb-root-ca-cert created
issuer.cert-manager.io/mongodb-root-ca-issuer created
configmap/namespace-scope created
statefulset.apps/icp-mongodb created
[INFO] Waiting for MongoDB copy to initialize
icp-mongodb-0   0/1   Init:1/2   0     10s
[INFO] Waiting for MongoDB copy to initialize
icp-mongodb-0   0/1   Init:1/2   0     21s
[INFO] Waiting for MongoDB copy to initialize
icp-mongodb-0   0/1   Init:1/2   0     31s
[INFO] Waiting for MongoDB copy to initialize
icp-mongodb-0   0/1   Init:1/2   0     41s
[INFO] Waiting for MongoDB copy to initialize
icp-mongodb-0   0/1   Init:1/2   0     51s
[INFO] Waiting for MongoDB copy to initialize
icp-mongodb-0   0/1   Init:1/2   0     62s
[INFO] Waiting for MongoDB copy to initialize
icp-mongodb-0   0/1   Init:1/2   0     72s
[INFO] Waiting for MongoDB copy to initialize
icp-mongodb-0   0/1   PodInitializing   0     82s
[INFO] Waiting for MongoDB copy to initialize
icp-mongodb-0   0/1   Running   0     92s
[✔] Temporary Mongo copy deployed to namespace preload
# Creating a PVC for the MongoDB dump
-----------------------------------------------------------------------
Now using project "keycloak" on server "https://api.installer-cp3pt0.cp.fyre.ibm.com:6443".
persistentvolumeclaim/cs-mongodump created
[✔] MongoDB PVC ready
# Backing up MongoDB in namespace keycloak
-----------------------------------------------------------------------
[INFO] Running Backup
job.batch/mongodb-backup created
mongodb-backup-mjv5f                                             0/1     ContainerCreating   0             0s
[INFO] Waiting for job pod mongodb-backup to complete
[✔] Job mongodb-backup completed in namespace keycloak

[INFO] Saving mongodb-backup logs in _mongodb-backup.log
[INFO] Deleting job mongodb-backup
job.batch "mongodb-backup" deleted
[✔] Backup Complete
# Moving restored mongodb volume to preload
-----------------------------------------------------------------------
persistentvolume/pvc-20ec7a2e-0672-4105-b469-036327f9330c patched
persistentvolume/pvc-20ec7a2e-0672-4105-b469-036327f9330c patched
The request is invalid: the server rejected our request due to an error in our request
persistentvolumeclaim "cs-mongodump" deleted
persistentvolumeclaim/cs-mongodump created
persistentvolume/pvc-20ec7a2e-0672-4105-b469-036327f9330c patched
[INFO] Waiting for pvc cs-mongodump to bind
persistentvolume/pvc-20ec7a2e-0672-4105-b469-036327f9330c patched (no change)
[INFO] Waiting for pvc cs-mongodump to bind
[✔] Restored MongoDB volume moved to namespace preload
# Restoring MongoDB to copy in namespace preload
-----------------------------------------------------------------------
[INFO] Running Restore
job.batch/mongodb-restore created
[INFO] Waiting for job pod mongodb-restore to complete
[✔] Job mongodb-restore completed in namespace preload

[INFO] Saving mongodb-restore logs in _mongodb-restore.log
[INFO] Deleting job mongodb-restore
job.batch "mongodb-restore" deleted
[✔] Restore Complete
# Deleting the stand up mongodb statefulset in preload
-----------------------------------------------------------------------
statefulset.apps "icp-mongodb" deleted
service "icp-mongodb" deleted
issuer.cert-manager.io "god-issuer" deleted
configmap "ibm-cpp-config" deleted
certificate.cert-manager.io "icp-mongodb-client-cert" deleted
configmap "icp-mongodb" deleted
configmap "icp-mongodb-init" deleted
configmap "icp-mongodb-install" deleted
secret "icp-mongodb-keyfile" deleted
secret "icp-mongodb-metrics" deleted
serviceaccount "ibm-mongodb-operand" deleted
service "mongodb" deleted
certificate.cert-manager.io "mongodb-root-ca-cert" deleted
issuer.cert-manager.io "mongodb-root-ca-issuer" deleted
configmap "namespace-scope" deleted
persistentvolume/pvc-20ec7a2e-0672-4105-b469-036327f9330c patched
persistentvolumeclaim "cs-mongodump" deleted
persistentvolume "pvc-20ec7a2e-0672-4105-b469-036327f9330c" deleted
[✔] MongoDB restored to new namespace preload
#  Copying secret platform-auth-idp-credentials from keycloak to preload 
secret/platform-auth-idp-credentials created
[✔] secret platform-auth-idp-credentials copied over to preload as platform-auth-idp-credentials
#  Copying secret platform-auth-ldaps-ca-cert from keycloak to preload 
secret/platform-auth-ldaps-ca-cert created
[✔] secret platform-auth-ldaps-ca-cert copied over to preload as platform-auth-ldaps-ca-cert
#  Copying configmap ibm-cpp-config from keycloak to preload 
configmap/ibm-cpp-config created
[✔] configmap ibm-cpp-config copied over to preload as ibm-cpp-config
#  Copying configmap common-web-ui-config from keycloak to preload 
configmap/common-web-ui-config created
[✔] configmap common-web-ui-config copied over to preload as common-web-ui-config
#  Copying configmap platform-auth-idp from keycloak to preload 
configmap/platform-auth-idp created
[✔] configmap platform-auth-idp copied over to preload as platform-auth-idp
#  Copying commonservice common-service from keycloak to preload 
commonservice.operator.ibm.com/preload-common-service-from-keycloak created
[✔] commonservice common-service copied over to preload as preload-common-service-from-keycloak
```
